report_qweb_element_page_visibility/views/layouts.xml

```diff
@@ -1,48 +1,50 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <openerp>
-<data>
+    <data>
 
-  <template id="minimal_layout" inherit_id="report.minimal_layout">
-    <xpath expr="//t[@t-if='subst is True']" position="replace">
-    	<t t-if="subst is True">
-        <script>
-            function subst() {
-                var vars = {};
-                var x = document.location.search.substring(1).split('&amp;');
-                for (var i in x) {
-                    var z = x[i].split('=', 2);
-                    vars[z[0]] = unescape(z[1]);
-                }
-                var x=['frompage', 'topage', 'page', 'webpage', 'section', 'subsection', 'subsubsection'];
-                for (var i in x) {
-                    var y = document.getElementsByClassName(x[i]);
-                    for (var j=0; j&lt;y.length; ++j)
-                        y[j].textContent = vars[x[i]];
+        <template id="minimal_layout" inherit_id="report.minimal_layout">
+            <xpath expr="//body" position="attributes">
+                <attribute name="onload">subst_wrapper();</attribute>
+            </xpath>
+            <xpath expr="//t[@t-if='subst is True']" position="after">
+                <script>
+                    function subst_wrapper() {
+                        if (typeof subst === "function") {
+                            subst();
+                        }
+                        var vars = {};
+                        var x = document.location.search.substring(1).split('&amp;amp;');
```

Comment:
you can avoid this kind of stuff by wrapping all javascript code in a [CDATA](https://en.wikipedia.org/wiki/CDATA) section

Reply:
Unfortunately, that made "wkhtmltopdf" not recognize the Javascript function anymore and thus not calling it.